File: AlgoExpert/cpp/palindrome_check.cc
/* Notes
Difficulty: Easy
*/

#include <string>

bool isPalindrome(std::string str) {
  // Write your code here.
  if (str.size() <= 1) return true;
  int left = 0, right = str.size() - 1;

  while (left <= right) {
    if (str[left++] != str[right--]) return false;
  }
  return true;
}

File: AlgoExpert/python/palindrome_check.py
"""
Difficulty: Easy
"""


def isPalindrome(string: str) -> bool:
    # Write your code here.
    if len(string) <= 1:
        return True

    left, right = 0, len(string) - 1

    while left <= right:
        if string[left] != string[right]:
            return False
        left += 1
        right -= 1
    return True
